### ISSUE TITLE
VersionAlreadyExistsException with ifExists=RETURN_OR_UPDATE

### ISSUE THREAD

**Issue by m-kay** [Apicurio/apicurio-registry#3555]
When using the maven plugin to publish a specific version and ifExists is configured with RETURN_OR_UPDATE I get a VersionAlreadyExistsException.

### Steps to Reproduce

1. Configure the maven plugin as following
```xml
<plugin>
	<groupId>io.apicurio</groupId>
	<artifactId>apicurio-registry-maven-plugin</artifactId>
	<version>2.4.3.Final</version>
	<executions>
		<execution>
			<phase>generate-sources</phase>
			<goals>
				<goal>register</goal>
			</goals>
			<configuration>
				<registryUrl>${SCHEMA_REGISTRY}/apis/registry/v2</registryUrl>
				<artifacts>
					<artifact>
						<artifactId>my-artifact</artifactId>
						<groupId>default</groupId>
						<version>1</version>
						<type>AVRO</type>
						<file>${project.basedir}/src/main/resources/avro/schema.avsc</file>
						<ifExists>RETURN_OR_UPDATE</ifExists>
						<canonicalize>true</canonicalize>
					</artifact>
				</artifacts>
			</configuration>
		</execution>
	</executions>
</plugin>
```
2. Execute the build -> artifact is published with version 1
3. Add a property to the schema.avsc
4. Execute the build again -> VersionAlreadyExistsException

### Expected behovior
I would have expected that the old version 1 gets deleted and my updated schema is published with version 1. In my case this would be necessary during development when something was missing in the schema after the first publishing.

**Comment by EricWittmann** [comment]
We discussed this issue today, and I think ultimately the issue here is some confusion about the meaning of `RETURN_OR_UPDATE`.  The operation you're invoking is a Create of an artifact with some content.  The `ifExists` param tells that operation what to do if the artifact already exists.  So the semantics of the `ifExists` param apply to the artifact, and so for `RETURN_OR_UPDATE` it means "return the information about the artifact if this content already exists, or else update the artifact (i.e. create a new version) if not".

If you include a version number for this operation, and the artifact already exists and has a version with the requested # but with different content, then it's impossible to update the artifact with a new version (because there is a version number clash).

I can see why you might have expected the version to be replaced, but our position is that versions should be immutable (as much as possible).  We do support deleting a version of an artifact, but that is a concession due to possible emergency production reasons (e.g. regulatory requirement in the case that content was registered that violates some policy).

We think the right thing to do for the use-case you've described (a property was added to the schema) is to create a new version (since a change was, in fact, made to the schema).

I would be interested in hearing more about a use-case where you want to change the schema and replace a version with new content...
